ciongke/cmd/hook/main.go

```diff
@@ -0,0 +1,337 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strconv"
+
+	gogithub "github.com/google/go-github/github"
+	"golang.org/x/oauth2"
+
+	"github.com/kubernetes/test-infra/ciongke/github"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/resource"
+	"k8s.io/kubernetes/pkg/apis/batch"
+	"k8s.io/kubernetes/pkg/client/restclient"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/labels"
+)
+
+var (
+	port         = flag.Int("port", 8888, "Port to listen on.")
+	jobImage     = flag.String("test-pr-image", "", "Image to create to test PRs.")
+	jobBucket    = flag.String("results-bucket", "", "Public GCS bucket to store results.")
+	sourceBucket = flag.String("source-bucket", "", "Private GCS bucket to store PR source code.")
+	namespace    = flag.String("namespace", "default", "Namespace for all CI objects.")
+	org          = flag.String("org", "", "GitHub org to trust.")
+	teamID       = flag.Int("team-id", 0, "GitHub team ID to trust.")
+	teamName     = flag.String("team-name", "", "GitHub team name that matches the ID.")
+
+	webhookSecretFile = flag.String("hmac-secret-file", "/etc/hmac/hmac", "Path to the file containing the GitHub HMAC secret.")
+	oauthSecretFile   = flag.String("oauth-secret-file", "/etc/oauth/oauth", "Path to the file containing the GitHub OAuth secret.")
```

Comment:
rename to githubTokenFile to match Github's language in https://github.com/settings/tokens